airbyte-webapp/src/components/connection/CatalogTree/next/StreamDetailsPanel/StreamFieldsTable/StreamFieldsTable.tsx

```diff
@@ -0,0 +1,197 @@
+
+  // prepare data for table
+  const tableData: TableStream[] = useMemo(
+    () =>
+      syncSchemaFields.map((stream) => ({
+        path: stream.path,
+        dataType: getDataType(stream),
+        cursorDefined: shouldDefineCursor && SyncSchemaFieldObject.isPrimitive(stream),
+        primaryKeyDefined: shouldDefinePk && SyncSchemaFieldObject.isPrimitive(stream),
+      })),
+    [shouldDefineCursor, shouldDefinePk, syncSchemaFields]
+  );
+
+  const columnHelper = createColumnHelper<TableStream>();
+
+  const sourceColumns = useMemo(
+    () => [
+      columnHelper.accessor("path", {
+        id: "sourcePath",
+        header: () => <FormattedMessage id="form.field.name" />,
+        cell: ({ getValue }) => pathDisplayName(getValue()),
+        meta: {
+          thClassName: styles.headerCell,
+          tdClassName: styles.textCell,
+        },
+      }),
+      columnHelper.accessor("dataType", {
+        id: "sourceDataType",
+        header: () => <FormattedMessage id="form.field.dataType" />,
+        cell: ({ getValue }) => (
+          <FormattedMessage id={`${getValue()}`} defaultMessage={formatMessage({ id: "airbyte.datatype.unknown" })} />
+        ),
+        meta: {
+          thClassName: styles.headerCell,
+          tdClassName: styles.dataTypeCell,
+        },
+      }),
+      columnHelper.accessor("cursorDefined", {
+        id: "sourceCursorDefined",
+        header: () => <FormattedMessage id="form.field.cursorField" />,
+        cell: ({ getValue, row }) =>
+          getValue() && (
+            <RadioButton checked={isCursor(row.original.path)} onChange={() => onCursorSelect(row.original.path)} />
+          ),
+        meta: {
+          thClassName: styles.headerCell,
+          tdClassName: styles.checkboxCell,
+        },
+      }),
+      columnHelper.accessor("primaryKeyDefined", {
+        id: "sourcePrimaryKeyDefined",
+        header: () => <FormattedMessage id="form.field.primaryKey" />,
+        cell: ({ getValue, row }) =>
+          getValue() && (
+            <CheckBox
+              checked={isPrimaryKey(row.original.path)}
+              onChange={() => onPkSelect(row.original.path)}
+              className={styles.checkbox}
+            />
+          ),
+        meta: {
+          thClassName: styles.headerCell,
+          tdClassName: styles.textCell,
+        },
+      }),
+    ],
+    [columnHelper, formatMessage, isCursor, isPrimaryKey, onCursorSelect, onPkSelect]
+  );
+
+  const destinationColumns = useMemo(
+    () => [
+      columnHelper.accessor("path", {
+        id: "destinationPath",
+        header: () => <FormattedMessage id="form.field.name" />,
+        cell: ({ getValue }) => pathDisplayName(getValue()),
+        meta: {
+          thClassName: styles.headerCell,
+          tdClassName: styles.textCell,
+        },
+      }),
+      // In the design, but we may be unable to get the destination data type
+      columnHelper.accessor("dataType", {
```

Comment:
The data type here is the source data type but not the destination data type. If we don't have a destination data type, we may need to remove this column.

Reply:
Sure, no pb! Will remove it